### DISCUSSION TITLE
Route53 module does not support adding an A record as an alias

### DISCUSSION THREAD

**Discussion by demimike** [gruntwork-io/knowledge-base#656] [Feedback]
[[terraform-aws-service-catalog](https://github.com/gruntwork-io/terraform-aws-service-catalog)/[modules](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/modules)/[networking](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/modules/networking/route53)
](https://github.com/gruntwork-io/terraform-aws-service-catalog/tree/master/modules/networking/route53)

Route53 module does not support adding an A record as an alias

In terraform it looks like:
resource "aws_route53_record" "www" {
  zone_id = aws_route53_zone.primary.zone_id
  name    = "example.com"
  type    = "A"

  alias {
    name                   = aws_elb.main.dns_name
    zone_id                = aws_elb.main.zone_id
    evaluate_target_health = true
  }

---

<ins datetime="2023-02-03T20:08:35Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109863">Tracked in ticket #109863</a></p>
</ins>

**Comment by gitsstewart** [comment]
@demimike thanks for the feedback. Can you also provide some context around why you'd like this functionality, perhaps what scenario is leading you to want to take this approach?

**Comment by zackproser** [comment]
Hi @demimike could you please share the exact error you're getting? Which version of the service catalog are you using? We'd also love to see the exact inputs you're configuring and how.

Thanks

**Comment by demimike** [comment]
We [image: alias.png]
are hosting many vanity domains for customers, and we would like to route
the all through cloudfront

**Comment by demimike** [comment]
It doesn't generate errors, I just can't configure A records as an alias.
We are currently using ver 99.2

**Comment by eak12913** [comment]
Hey @demimike we are currently missing that feature in the Route53 Service Catalog service but have filed an issue to see if we can get it added sooner rather than later. I obviously can't commit to a specific date, but I can commit to discussing this with our team and getting a ball park answer.

**Comment by oredavids** [comment]
PR for this is currently under review: https://github.com/gruntwork-io/terraform-aws-service-catalog/pull/1829

**Reply by oredavids** [reply]
This is now available from [v0.102.5](https://github.com/gruntwork-io/terraform-aws-service-catalog/releases/tag/v0.102.5) of the service catalog's Route53 module.
